Miscellaneous refactoring

openjck opened the pull request.

openjck committed:
Remove one-line parseDiff function

openjck committed:
Add comments, change spacing in getConfig

openjck committed:
Return detailed commits from getPullRequestCommits

openjck committed:
Minor cleanup in addPullRequestComments

openjck committed:
Remove one-line trackUsage function

openjck committed:
Refactor parseCSS, renderComment, and sourceParse

openjck committed:
Use logger.error correctly, rewrite messages

openjck committed:
Beautify gulpfile-prod.js

darkwing commented on processor.js:
```diff
@@ -0,0 +1,68 @@
+            onFeatureUsage: handleIncompatibility
+        })).process(fileContents, {
+            from: '/' + file.filename
+        }).then(function(response) {});
```
I think we should eliminate this line.  I like that we put the `then` there as a reminder we *can* do something here, but since we aren't, we should axe for now.

openjck replied:
Ah definitely. I meant to nix that. Thanks!

openjck replied:
As it turns out, `.then()` does need to be there for commenting to work. The body can be empty though. Not sure why, but I seem to recall Blixa saying something about it.

darkwing replied:
...whaaaaa

darkwing commented on processor.js:
```diff
@@ -0,0 +1,68 @@
+            sourceMapConsumer = new sourceMap.SourceMapConsumer(compiler.sourcemap);
+
+            function handleStylusIncompatibility(incompatibility) {
+                var compiledPosition = sourceMapConsumer.originalPositionFor(incompatibility.usage.source.start);
```
Since we use `incompatibility.usage.source.start` in 3 lines, we should create a shortcut reference var for it.

openjck replied:
Sounds good to me!

darkwing commented:
One recommendation I have is, at the top of each function, writing a summary of what the function is meant to do.  At this point I feel like this project's code is under-documented;  with explanations of each function, contributing will be easier.

darkwing commented on hook.js:
```diff
@@ -39,16 +35,36 @@ function handle(request, response) {
+            currentCommit.files.forEach(function(file) {
+
+                // And process the files
+
```
Extra line?

openjck replied:
I added the extra line just to show that the comment applies to the entire body. I can add it to line 48.

darkwing commented on hook.js:
```diff
@@ -39,16 +35,36 @@ function handle(request, response) {
+                    commenter.postPullRequestComment(commentURL, comment, file.filename, currentCommit.sha, line);
+                }
+
+                switch (path.extname(file.filename).toLowerCase()) {
```
The `css` and `styl` functions have the same signature.  If `sass` and `less` will have the same signature, we should just have this switch return the processing function only.

openjck replied:
Great call. Will do.

openjck replied:
Done! :smile:

This thread is still here because a different line was changed.

openjck commented:
> One recommendation I have is, at the top of each function, writing a summary of what the function is meant to do. At this point I feel like this project's code is under-documented; with explanations of each function, contributing will be easier.

Definitely. Will do.

darkwing commented on processor.js:
```diff
@@ -0,0 +1,68 @@
+'use strict';
```
Much, much better file name

openjck replied:
:tada:

darkwing commented on hook.js:
```diff
@@ -80,79 +95,46 @@ function getConfig(repo, branch) {
+
         if (error) {
-            deferred.reject('Fetching commits failed: ' + error);
+            deferred.reject('Error fetching pull request commits:', error);
```
Do we want to add more info here, like the project or PR, just for the sake of future debugging?

openjck replied:
Definitely. I'll add that. Thanks!

openjck committed:
Stop logging full payload

We added this in #55. We do a better job logging errors now, and the
full payload tends to clutter the log. GitHub also shows it on the
*Webhooks & Services* page.

darkwing commented on processor.js:
```diff
@@ -0,0 +1,70 @@
+
+            if (error) return logger.error('Error compiling Stylus of ' + repo + ' / ' + branch + ' / ' + file.filename + ':', error);
+
+            sourceMapConsumer = new sourceMap.SourceMapConsumer(compiler.sourcemap);
```
This and its initial var declaration can be moved into handleStylusIncompatibility

openjck replied:
Definitely. Will do!

openjck committed:
Improve comments, do some minor refactoring

darkwing commented:
R+, nice!
